File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/TimePeriod.kt
package com.radixdlt.sargon.extensions

import com.radixdlt.sargon.TimePeriod
import com.radixdlt.sargon.newTimePeriodWithDays
import com.radixdlt.sargon.timePeriodToDays

fun TimePeriod.Companion.withDays(value: Int): TimePeriod = newTimePeriodWithDays(value.toUShort())

val TimePeriod.days
    get() = timePeriodToDays(this)

File: crates/uniffi/uniffi_SPLIT_ME/src/profile/mfa/security_structures/models/time_period.rs
use crate::prelude::*;
use sargon::TimePeriod as InternalTimePeriod;

/// Time period unit expressed in days, weeks, or years.
///
/// Used to represent in the hosts UI the time period.
#[derive(
    Clone, Debug, Copy, PartialEq, Eq, Hash, InternalConversion, uniffi::Record,
)]
pub struct TimePeriod {
    /// The value of the time period.
    pub value: u16,
    /// The unit of the time period.
    pub unit: TimePeriodUnit,
}

#[uniffi::export]
pub fn new_time_period_sample() -> TimePeriod {
    InternalTimePeriod::sample().into()
}

#[uniffi::export]
pub fn new_time_period_sample_other() -> TimePeriod {
    InternalTimePeriod::sample_other().into()
}

#[uniffi::export]
pub fn new_time_period_with_days(value: u16) -> TimePeriod {
    InternalTimePeriod::with_days(value).into()
}

#[uniffi::export]
pub fn time_period_to_days(time_period: &TimePeriod) -> u16 {
    time_period.into_internal().days()
}
